### PR TITLE
Gyr1 549 remove ctc specific code from the efile submission state machine

### PR DESCRIPTION
**Pull request by mrotondo**
## Link to pivotal/JIRA issue
- https://codeforamerica.atlassian.net/browse/GYR1-549
## Is PM acceptance required?
- No - merge after code review approval
## What was done?
- CTC-specific code is removed from the efile submission state machine
- I left the checks for statefile-ness intact - we can get rid of those later
- I deleted some messages that had been moved into state file directories despite there not being a plan to send them to state file users
## How to test?
- Unit tests should cover this change

### CODE REVIEW

File: spec/controllers/ctc/portal/submission_pdfs_controller_spec.rb
Comment (mrotondo) [inline spec/controllers/ctc/portal/submission_pdfs_controller_spec.rb:1]: revisit this

File: spec/lib/seeder_spec.rb
Comment (mrotondo) [inline spec/lib/seeder_spec.rb:1]: we want at least some test data for manual testing

Comment (mrotondo) [reply spec/lib/seeder_spec.rb:1]: or examine the need for this test case

File: spec/models/efile_submission_spec.rb [hunk 1]
```diff
@@ -340,55 +284,15 @@
...
-        allow(address_double).to receive(:valid?).and_return false
-
-      end
-
-      it "tries again up to 3x and returns the address object if it fails after 3 tries" do
-        expect(StandardizeAddressService).to receive(:new).exactly(3).times
-        expect(submission.generate_verified_address).to eq address_double
-      end
-    end
+    # TODO(state-file)
```
Comment (mrotondo) [inline spec/models/efile_submission_spec.rb:291]: delete this ?

File: spec/models/efile_submission_spec.rb [hunk 2]
```diff
@@ -397,126 +301,47 @@
...
-            submission = previous_submission.tax_return.efile_submissions.last
+            submission = previous_submission.data_source.efile_submissions.last
             expect(submission.admin_resubmission?).to eq false
           end
         end
       end
     end
   end
 
-  describe "#generate_filing_pdf" do
```
Comment (mrotondo) [inline spec/models/efile_submission_spec.rb:438]: might need to keep

File: spec/services/efile/poll_for_acknowledgments_service_spec.rb [hunk 3]
```diff
@@ -97,15 +97,18 @@
...
 
         context "when the IRS has an acknowledgement ready for this submission" do
           before do
             allow(Efile::GyrEfilerService).to receive(:run_efiler_command)
-                                                .with("test", "acks", state_efile_submission.irs_submission_id)
+                                                .with("test", "acks", ready_for_ack_state_efile_submission.irs_submission_id)
+                                                .and_return expected_irs_return_value
+            allow(Efile::GyrEfilerService).to receive(:run_efiler_command)
+                                                .with("test", "submissions-status", transmitted_state_efile_submission.irs_submission_id)
                                                 .and_return expected_irs_return_value
```
Comment (mrotondo) [inline spec/services/efile/poll_for_acknowledgments_service_spec.rb:112]: this one should probably return ""

File: app/jobs/state_file/after_transition_tasks_for_rejected_return_job.rb
[the remark this answers is not in the document]
Comment (mrotondo) [reply app/jobs/state_file/after_transition_tasks_for_rejected_return_job.rb:1]: decided this was unnecessary

File: app/lib/seeder.rb
[the remark this answers is not in the document]
Comment (mpidcock) [reply app/lib/seeder.rb:1]: added back with some adjustments

File: app/models/efile_submission.rb [hunk 4]
```diff
@@ -159,7 +151,7 @@ def previously_transmitted_submission
   end
 
   def source_record
-    is_for_state_filing? ? data_source : tax_return
+    data_source
```
[the remark this answers is not in the document]
Comment (mpidcock) [reply app/models/efile_submission.rb:154]: done

File: app/models/efile_submission.rb [hunk 5]
```diff
@@ -192,50 +184,19 @@ def pdf_bundle_filename
...
+    output_file
   end
 
   def manifest_class
-    if is_for_state_filing?
-      return SubmissionBuilder::StateManifest
-    end
-
-    SubmissionBuilder::FederalManifest
+    SubmissionBuilder::StateManifest
```
[the remark this answers is not in the document]
Comment (mpidcock) [reply app/models/efile_submission.rb:195]: no, prefer it this way

File: spec/controllers/ctc/portal/submission_pdfs_controller_spec.rb
Comment (mrotondo) [reply spec/controllers/ctc/portal/submission_pdfs_controller_spec.rb:1]: done - brought the contents of this test back and fixed pdf generation.

File: spec/lib/seeder_spec.rb
Comment (mrotondo) [reply spec/lib/seeder_spec.rb:1]: brought back the seeder data & restored the test

File: spec/models/efile_submission_spec.rb (hunk 1 above)
Comment (mrotondo) [reply spec/models/efile_submission_spec.rb]: done

File: spec/models/efile_submission_spec.rb (hunk 2 above)
Comment (mrotondo) [reply spec/models/efile_submission_spec.rb]: restored this and the address verification test

File: spec/services/efile/poll_for_acknowledgments_service_spec.rb (hunk 3 above)
Comment (mrotondo) [reply spec/services/efile/poll_for_acknowledgments_service_spec.rb]: done

File: app/state_machines/efile_submission_state_machine.rb [hunk 6]
```diff
@@ -79,6 +78,9 @@ class EfileSubmissionStateMachine
 
   after_transition(to: :rejected, after_commit: true) do |submission, transition|
     StateFile::AfterTransitionTasksForRejectedReturnJob.perform_later(submission, transition)
+
+
+
     EfileSubmissionStateMachine.send_mixpanel_event(submission, "state_file_efile_return_rejected")
```
Comment (mpidcock) [inline app/state_machines/efile_submission_state_machine.rb:84]:
```suggestion
    StateFile::AfterTransitionTasksForRejectedReturnJob.perform_later(submission, transition)
    EfileSubmissionStateMachine.send_mixpanel_event(submission, "state_file_efile_return_rejected")
```

**Review by mpidcock — APPROVED** [review]
✂️

**Pull request merged**